lib/oli/delivery/sections.ex

```diff
@@ -1630,69 +1613,41 @@ defmodule Oli.Delivery.Sections do
       ]
   """
   def get_ordered_container_labels(section_slug, opts \\ []) do
-    short_label = opts[:short_label] || false
-
-    label_for = fn c ->
-      container_label =
-        get_container_label(
-          c.numbering_level,
-          c.customizations || Map.from_struct(CustomLabels.default())
-        )
+    SectionCache.get_or_compute(section_slug, :ordered_container_labels, fn ->
+      fetch_ordered_container_labels(section_slug, opts)
+    end)
+  end
 
-      if short_label do
-        ~s{#{container_label} #{c.numbering_index}}
-      else
-        ~s{#{container_label} #{c.numbering_index}: #{c.title}}
-      end
-    end
+  def fetch_ordered_container_labels(section_slug, opts \\ []) do
+    short_label = opts[:short_label] || false
 
-    ordered_containers =
-      SectionCache.get_or_compute(section_slug, :ordered_containers, fn ->
-        fetch_ordered_containers(section_slug)
-      end)
+    # TODO: OPTIMIZATION replace this with a minimal hierarchy query after v26.2 is merged
+    %Section{customizations: customizations} = get_section_by_slug(section_slug)
+    full_hierarchy = DeliveryResolver.full_hierarchy(section_slug)
 
-    ordered_containers
-    |> Enum.map(fn {id, c} ->
-      {id, label_for.(c)}
+    full_hierarchy
+    |> Hierarchy.flatten()
+    |> Enum.filter(fn node ->
+      node.revision.resource_type_id == ResourceType.get_id_by_type("container")
+    end)
+    |> Enum.map(fn %HierarchyNode{resource_id: resource_id, section_resource: sr, revision: rev} ->
+      {resource_id,
+       label_for(sr.numbering_level, sr.numbering_index, rev.title, short_label, customizations)}
     end)
   end
 
-  defp fetch_ordered_containers(section_slug) do
-    container_type_id = Oli.Resources.ResourceType.get_id_by_type("container")
+  defp label_for(numbering_level, numbering_index, title, short_label, customizations) do
+    container_label =
+      get_container_label(
+        numbering_level,
+        customizations || Map.from_struct(CustomLabels.default())
+      )
 
-    SectionResource
```

Comment:
This logic of using a single query and numbering level/index was not quite right. The only way to actually get the correct ordering according to the hierarchy is to load the actual hierarchy and use it to sort